src/python/CMSSpark/data_aggregation.py

```diff
@@ -0,0 +1,385 @@
+            dest="verbose", default=False, help="Verbose output")
+        self.parser.add_argument("--fout", action="store",
+            dest="fout", default="", help='Output directory path')
+
+
+def run_agg_jm(date, fout, ctx, sql_context, verbose=False):
+
+    if verbose:
+        print 'Starting JobMonitoring part'
+
+    # Convert date
+    date = long_date_string(date)
+
+    # Create JobMonitoring tables in sql_context
+    jm_df = jm_tables(ctx, sql_context, date=date, verbose=verbose)
+
+    if verbose:
+        print 'Found ' + str(jm_df['jm_df'].count()) + ' records in JobMonitoring stream'
+
+    # - site name                +
+    # - dataset name             +
+    # - number of access (nacc)  +
+    # - distinct users           +
+    # - stream: crab             +
+
+    cols = ['SiteName AS site_name',
+            'dataset_name',
+            'count(dataset_name) AS nacc',
+            'count(distinct(UserId)) AS distinct_users',
+            '\"crab\" as stream']
+
+    # Build a query with "cols" columns
+    query = ("SELECT %s FROM jm_df "\
+             "JOIN f_b_s_df ON f_b_s_df.file_name = jm_df.FileName "\
+             "GROUP BY jm_df.SiteName, dataset_name") % ','.join(cols)
+
+    result = run_query(query, sql_context, verbose)
+
+    result = result.sort(desc("nacc"))
+
+    if verbose:
+        print 'Finished JobMonitoring part (output is ' + str(result.count()) + ' records)'
+
+    return result
+
+
+def run_agg_eos(date, fout, ctx, sql_context, verbose=False):
+
+    if verbose:
+        print 'Starting EOS part'
+
+    date = short_date_string(date)
+
+    # Create EOS tables in sql_context
+    eos_df = eos_tables(sql_context, date=date, verbose=verbose)
+
+    if verbose:
+        print 'Found ' + str(eos_df['eos_df'].count()) + ' records in EOS stream'
+
+    # - site name                +
+    # - dataset name             +
+    # - number of access (nacc)  +
+    # - distinct users           +
+    # - stream: eos              +
+
+    cols = ['site_name',
+            'dataset_name',
+            'count(dataset_name) AS nacc',
+            'count(distinct(eos_df.user_dn)) AS distinct_users',
+            '\"eos\" as stream']
+
+
+    # Build a query with "cols" columns
+    query = ("SELECT %s FROM eos_df " \
+             "JOIN f_b_s_df ON f_b_s_df.file_name = eos_df.file_lfn " \
+             "GROUP BY site_name, dataset_name") % ','.join(cols)
+
+    result = run_query(query, sql_context, verbose)
+
+    result = result.sort(desc("nacc"))
+
+    if verbose:
+        print 'Finished EOS part (output is ' + str(result.count()) + ' records)'
+
+    return result
+
+
+def run_agg_aaa(date, fout, ctx, sql_context, verbose=False):
+
+    if verbose:
+        print 'Starting AAA part'
+
+    # Convert date
+    date = short_date_string(date)
+
+    # Use enr instead of raw files
+    hdir = 'hdfs:///project/monitoring/archive/xrootd/enr/gled'
```

Comment:
do not use hardcoded values, pass the hdir as an agrument to this function, you may assign default value for this argument though.